internal/interface/crawler/crawler.go

```diff
@@ -34,10 +48,39 @@ func NewCrawler(db *sql.DB, articlesRepo repository.ArticleRepo) *Crawler {
 }
 
 type Configuration struct {
-	RootURL             string
+	RootURL			 string
 	DesiredArticleCount int
 }
 
+func (c *Crawler) GetUnvisitedURL() (string, error) {
+	rows, err := c.db.Query("SELECT URL FROM CrawlStatus WHERE Visited = false LIMIT 1;")
```

Comment:
Then all of a sudden, another team member arrives and changes this: getURLConcurrency      = 1 ... (o-oh!((

Reply:
Generally it's a good idea to change status if some piece of work is taken by some agent. 

We could assume that there is only one agent, but it is a dangerous assumption in case of future possible horizontal scaling.

Reply:
Oh, I just realized that there is a concurrency problem event with one worker. We should discuss it tomorrow.